src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -719,7 +717,7 @@ ExecuteSingleModifyTask(CitusScanState *scanState, Task *task, bool expectResult
 	 * customers already use functions that touch multiple shards from within
 	 * a function, so we'll ignore functions for now.
 	 */
-	if (IsTransactionBlock())
+	if (IsTransactionBlock() || multipleTasks)
```

Comment:
Hm, arguably that'd be better done on the callsite? But I don't care that much...